[github] Tests reports

vguerci opened the pull request.

TurpIF commented on .github/workflows/main.yml:
```diff
@@ -60,3 +60,16 @@ jobs:
         bundle exec fastlane test
+
+    - name: Test reports upload
+      uses: actions/upload-artifact@master
```
upload-artifact is now using main ad default branch

TurpIF commented on .github/workflows/main.yml:
```diff
@@ -60,3 +60,16 @@ jobs:
+        path: "fastlane/test_output"
+
+    - name: Tests reports annotations
+      uses: ashley-taylor/junit-report-annotations-action@1.3
```
FYI this action is currently broken. This [PR](https://github.com/ashley-taylor/junit-report-annotations-action/pull/16) should fix it.

vguerci commented:
Removed annotations until It works, fixed the upload version.
Side note, you should avoid targeting @main / @master as it is fragile and could even be, even if unlikely, an attack vector.

vguerci commented:
Switched to main as it is not v4 specific _(and also because tests are failing on v4 🤔)_
